### PR TITLE
mysql: add foreign key error

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
Add error msg

### CODE REVIEW

File: mysql/errcode.go [hunk 1]
```diff
@@ -939,6 +939,7 @@ const (
 	ErrSpatialFunctionalIndex                                       = 3760
 	ErrWrongKeyColumnFunctionalIndex                                = 3761
 	ErrFunctionalIndexOnField                                       = 3762
+	ErrReferencedForeignKey                                         = 3780
```
Comment (tangenta) [inline mysql/errcode.go]: According to the MySQL [document](https://dev.mysql.com/doc/refman/8.0/en/server-error-reference.html),
> Error number: 3780; Symbol: ER_FK_INCOMPATIBLE_COLUMNS; SQLSTATE: HY000
Message: Referencing column '%s' and referenced column '%s' in foreign key constraint '%s' are incompatible.

The error name should be `ErrFKIncompatibleColumns`.

Comment (crazycs520) [reply mysql/errcode.go]: great, done.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by kennytm** [comment]
(CLA is borked)

**Pull request merged**